src/os/bluestore/BlueStore.cc

```diff
@@ -1630,10 +1653,19 @@ bool BlueStore::ExtentMap::update(Onode *o, KeyValueDB::Transaction t,
 	assert(p->shard_info->offset == p->offset);
 	p->shard_info->bytes = len;
 	p->shard_info->extents = nn;
-	t->set(PREFIX_OBJ, p->key, bl);
+        
+        encoded_shards[pos].key = &p->key;
+        dirty_shards.push_back(encoded_shards[pos]);
 	p->dirty = false;
       }
       p = n;
+      ++pos;
+    }
+    //schedule DB update for dirty shards
+    auto it = dirty_shards.begin();
+    while( it != dirty_shards.end()) {
+      t->set(PREFIX_OBJ, *(it->key), it->bl);
```

Comment:
And a small bonus of my approach - we don't need to enumerate ALL the shards twice. I'm iterating over dirty blobs only at the final stage.